File: palin.pas
//Đếm số là Palindrome <=N và chia hết cho m
var n, m: longint;
    i, total: longint;
function reverse_digits(original_number:longint):longint;
var aux, reversed_number: longint;
begin
 reversed_number := 0;
 while (original_number <> 0) do
  begin
   reversed_number := reversed_number * 10 + original_number mod 10;
   original_number := original_number div 10;
  end;
  reverse_digits := reversed_number;
end;
 
function is_palindrome(number:longint):boolean;
begin
 if (number = reverse_digits(number)) then is_palindrome := true
 else is_palindrome := false;
end;
 
begin
	readln(n, m);
	total := 0;
	for i := 1 to n do 
		if (is_palindrome(i)) and (i mod m = 0) then total:=total+1;
	write(total);
end.

File: palin.cpp
//Đếm số là Palindrome <=N và chia hết cho m
#include <bits/stdc++.h>
using namespace std;
int oneDigit(int num)
{
    return (num >= 0 && num < 10);
}
bool isPalUtil(int num, int* dupNum)
{
    if (oneDigit(num))
        return (num == (*dupNum) % 10);
    if (!isPalUtil(num/10, dupNum))
        return false;
    *dupNum /= 10;
    return (num % 10 == (*dupNum) % 10);
}
int isPal(int num)
{
    if (num < 0)
       num = -num;
    int *dupNum = new int(num);
    return isPalUtil(num, dupNum);
}
int main() {
	int n, m, total = 0;
	cin >> n >> m;
	for(int i=1; i<=n; ++i)
		if(isPal(i) && i%m==0)
			total++;
	cout << total;
	return 0;
}
